src/components/TimeSeriesCard/TimeSeriesCard.jsx

```diff
@@ -1,124 +1,436 @@
-      : range.min;
-  const max = range === undefined ? undefined : range.max ? range.max : moment().toISOString();
-  const chartData = !isEmpty(values)
-    ? series.label === undefined
-      ? {
-          xFormat: '%Y-%m-%dT%H:%M:%S.%LZ',
-          colors: series.reduce(
-            (acc, curr) =>
-              Object.assign({}, acc, curr.color ? { [curr.dataSourceId]: curr.color } : {}),
-            {}
-          ),
-          names: series.reduce(
-            (acc, curr) =>
-              Object.assign({}, acc, curr.label ? { [curr.dataSourceId]: curr.label } : {}),
-            {}
-          ),
-          json: values,
-          keys: { value: series.map(line => line.dataSourceId), x: timeDataSourceId },
-          type: 'line',
-        }
-      : {
-          // a single dataset
-          x: timeDataSourceId,
-          xFormat: '%Y-%m-%dT%H:%M:%S.%LZ',
-          colors: {
-            [series.dataSourceId]: series.color,
+class TimeSeriesCard extends Component {
+  constructor(props) {
+    super(props);
+    // Create the ref
+    this.c3ChartComponent = React.createRef();
+    this.state = {
+      colors: {},
+    };
+  }
+
+  // componentDidUpdate(prevProps) {
+  //   // Typical usage (don't forget to compare props):
+  //   console.log('size', this.props);
+  //   if (this.props.size !== prevProps.size) {
+  //     console.log('force resize?');
+  //     this.c3ChartComponent.chart.flush();
+  //   }
+  // }
+
+  componentDidMount() {
+    this.setState({
+      colors: this.c3ChartComponent.chart ? this.c3ChartComponent.chart.data.colors() : {},
+    });
+  }
+
+  customTooltip = (d, defaultTitleFormat, defaultValueFormat, color) => {
+    const content = (
+      <Fragment>
+        <StyledTooltipCaret />
+        {d.map((item, index) => {
+          const dateFormat = new Date(item.x);
+          const titleX = dateFormat.toLocaleDateString('default', {
+            year: 'numeric',
+            month: 'short',
+            day: 'numeric',
+            hour: 'numeric',
+            minute: 'numeric',
+            second: 'numeric',
+          });
+          const valueformat = defaultValueFormat(item.value, item.id, item.index);
+          return (
+            <Fragment>
+              {d.length > 1 ? (
+                <Fragment>
+                  {index === 0 ? <StyledTooltipTitle>{titleX}</StyledTooltipTitle> : null}
+                  <StyledTooltipText color={color(item.id)}>
+                    {item.name}: <StyledTooltipValue>{valueformat}</StyledTooltipValue>
+                  </StyledTooltipText>
+                </Fragment>
+              ) : (
+                <Fragment>
+                  <StyledTooltipText>{titleX}</StyledTooltipText>
+                  <StyledTooltipValue>Value: {valueformat}</StyledTooltipValue>
+                </Fragment>
+              )}
+            </Fragment>
+          );
+        })}
+      </Fragment>
+    );
+
+    return ReactDOMServer.renderToString(content);
```

Comment:
This is very interesting!  I assume you had to do this because this is what C3 accepts as a tooltip, but it took me a while of thinking about it to figure it out.  Can you add a comment to explain the 'why' of it?